src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -298,6 +376,9 @@ private void VisitFieldOrPropertyInitializer(BoundInitializer initializer)
         {
             using var _ = new LocalScope(this, node.Locals);
 
+            this.Visit(node.DeclarationsOpt);
```

Comment:
Out-of-scope for this PR, but this made me wonder whether the placeholder discipline could be enforced/factored into the base type, so it would apply to all our bound node visitors.